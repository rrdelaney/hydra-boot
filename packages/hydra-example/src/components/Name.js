import React from 'react'
import { connect } from 'react-redux'
import { Header } from 'semantic-ui-react'

const mapStateToProps = state => ({
  name: state.auth && state.auth.name
})

export function Name({ name }) {
  return name ? (
    <Header as="h2">Hello {name}!!</Header>
  ) : (
    <Header as="h2">Not logged in</Header>
  )
}

export default connect(mapStateToProps)(Name)
