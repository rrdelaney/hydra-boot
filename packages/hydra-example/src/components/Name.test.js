import React from 'react'
import { mount } from 'enzyme'
import { Name } from './Name'

test('Can render', () => {
  const wrapper = mount(<Name name="ryan" />)

  expect(wrapper).toMatchSnapshot()
})
