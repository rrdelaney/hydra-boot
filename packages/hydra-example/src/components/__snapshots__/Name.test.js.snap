// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Can render 1`] = `
<Name
  name="ryan"
>
  <Header
    as="h2"
  >
    <h2
      className="ui header"
    >
      Hello 
      ryan
      !!
    </h2>
  </Header>
</Name>
`;
